### PR TITLE
Proto 1.4 update

### PR DESCRIPTION
**Pull request by McSpider**
Adds documentation for the players (FFA), item mods and crafting modules.
Also includes cleanup of the kit item/armor examples and `<time>` limit elements.

### CODE REVIEW

File: src/modules/crafting.haml [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+                                %td
+                                    A custom smelting recipe.
+                                    %a.left-ref-link{:href=>"#smelt"} <i class="fa fa-chevron-down"></i>
+                                %td
+                                    %span.label.label-default Smelt Recipe Sub-elements
+                            %tr
+                                %td
+                                    %pre
+                                        %code= html_escape('<disable>')
+                                %td Disable all recipes for this item.
```
Comment (jedediah) [inline src/modules/crafting.haml]: "Disable all vanilla recipes for this item"

File: src/modules/crafting.haml [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+                            %tr
+                                %th Attribute
+                                %th Description
+                                %th Value
+                        %tbody
+                            %tr
+                                %td
+                                    %code override
+                                %td
+                                    Override all vanilla recipes resulting in this material and data.
```
Comment (jedediah) [inline src/modules/crafting.haml]: "Disable all vanilla recipes for the same item. This is just a convenient alternative to the <disable> element."

**Comment by matic0basle** [comment]
Just wondering, why the removal of the Team Deathmatch page?

**Comment by mrapple** [comment]
Not really relevant to this PR. My guess is because it uses the `score` module which is already covered under gamemodes. All the other gamemodes have their own distinct modules.

File: src/modules/crafting.haml [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+                        <crafting>
+                            <!-- Add custom shaped recipe and disable vanilla recipes for the same item. -->
+                            <shaped override="true">
+                                <result>stone pickaxe</result>
+                                <shape>
+                                    <row>FFF</row>
+                                    <row>.S.</row>
+                                    <row>.S.</row>
+                                </shape>
+                                <ingredient symbol="F">flint</ingredient >
```
Comment (jedediah) [inline src/modules/crafting.haml]: Mention that material:data format is supported

File: src/modules/crafting.haml [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+                                %td
+                                    %span.label.label-danger Required
+                                    %span.label.label-warning Unique
+                                    An ingredient used in this recipe.
+                                    %br
+                                    %i Only one shape per recipe is allowed.
+                                %td
+                                    %pre
+                                        %code= html_escape('<row>')
+                            %tr
```
Comment (jedediah) [inline src/modules/crafting.haml]: Mention that material:data format is supported

File: src/modules/crafting.haml [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+                        %thead
+                            %tr
+                                %th Element
+                                %th Description
+                                %th Value/Children
+                        %tbody
+                            %tr
+                                %td
+                                    %pre
+                                        %code= html_escape('<result>')
```
Comment (jedediah) [inline src/modules/crafting.haml]: This uses the same format as `<item>` in kits, so we should use the `material="..."` attribute and mention that all kit item features are supported.

File: src/modules/item_mods.haml [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+            = @page_title = "Modules"
+            %p.lead Game Settings
+    %section
+        .row
+            = partial('modules_sidebar')
+            .col-sm-9
+                .page-header
+                    %h1= @page_sub_title = "Item Mods"
+                :markdown
+                    The item mods module allows modification of all items present on a map or created during the match. This can be used, for example, to make all gold swords unbreakable or give all items of a specify type a custom name.
```
Comment (jedediah) [inline src/modules/item_mods.haml]: specify -> specific

File: src/modules/players.haml [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+                                    %code min
+                                %td
+                                    Minimum amount of players required.
+                                %td
+                                    %span.label.label-primary Number
+                            %tr
+                                %td
+                                    %code max
+                                %td
+                                    Maximum amount of players. (soft limit)
```
Comment (jedediah) [inline src/modules/players.haml]: Let's fix this before making another copy of it. This one can just say "Maximum number of players that can play in the match."

Comment (jedediah) [inline src/modules/players.haml]: Actually, how about this:

"Player limit -- normal players cannot join the match once it reached this size."

File: src/modules/players.haml [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+                                    %br
+                                    %i Players with certain privileges may join over this limit till <code>max-overfill</code> is reached.
+                                %td
+                                    %span.label.label-primary Number
+                            %tr
+                                %td
+                                    %code max-overfill
+                                %td Maximum player count hard limit. Must be greater than the defined <code>max</code>.
```
Comment (jedediah) [inline src/modules/players.haml]: "Player overfill -- premium players cannot join the match once it reaches this size."

File: src/modules/proto.haml [hunk 9]
```diff
@@ -73,10 +69,13 @@
...
                         ```
-
                     * A team can have a minimum required size with the `min=""` attribute.
                     * A players gamemode can be changed by using a the `<game-mode>adventure</game-mode>` kit.
-
+                    * New world border module, that can be used to create a shifting and/or auto resizing world border.
+                    * Support added for block punch and trample block drops.
+                    * Custom item crafting recipe module.
+                    * Modify all items of a specific type for the whole match.
+                    * Team less gamemodes with the `<players/>` module.
```
Comment (jedediah) [inline src/modules/proto.haml]: "Teamless"

File: src/modules/proto.haml [hunk 10]
```diff
@@ -73,10 +69,13 @@
                         <!-- Other modules. -->
                         </map>
                         ```
-
                     * A team can have a minimum required size with the `min=""` attribute.
                     * A players gamemode can be changed by using a the `<game-mode>adventure</game-mode>` kit.
-
+                    * New world border module, that can be used to create a shifting and/or auto resizing world border.
```
Comment (jedediah) [inline src/modules/proto.haml]: We should decide what exactly is going to be listed on this page. Until now, it has just been breaking changes. If this is going to be a complete changelog, then we should have a seperate list of breaking changes.

Comment (McSpider) [inline src/modules/proto.haml]: It is already listing some none-breaking changes, a separate list or labeling the changes that are breaking and moving them to the top of the change list would work.

File: src/modules/regions.haml [hunk 11]
```diff
@@ -145,7 +145,7 @@
                     <br/>
                     ### Applying Things to Regions
 
-                    Filters, kits, and velocity changes can be applied to regions by using an `<apply>` element. The apply element can contain more than one region. If the apply element contains more than one region the same settings applied to all of them. Applies can also be defined in the filters element.
+                    Filters, kits, and velocity changes can be applied to regions by using an `<apply>` element. The apply element can contain more than one region. If the apply element contains more than one region the same settings are applied to all of them, applies with no region default to `<everywhere/>` Applies can also be defined in the filters element.
```
Comment (jedediah) [inline src/modules/regions.haml]: This is a seperate sentence.

File: src/modules/teams.haml [hunk 12]
```diff
@@ -10,7 +10,7 @@
                 .page-header
                     %h1= @page_sub_title = "Teams"
                 :markdown
-                    Teams are a basic and important component of almost all matches, this module allows you to setup their attributes.
+                    Teams are a basic and important component of almost all matches, this module allows you to setup their attributes. Matches without teams can be setup with the [players](/modules/players) module.
```
Comment (jedediah) [inline src/modules/teams.haml]: We can remove the thing about almost all matches having teams.

**Comment by matic0basle** [comment]
@mrapple Only just noticed it in this. Thanks.

**Comment by McSpider** [comment]
@jedediah I've fixed all the noted issues, let me know what you think about the updated proto page.

**Comment by jedediah** [comment]
Looks great, merge it up

**Pull request merged**